src/sink/mod.rs

```diff
@@ -6,64 +6,75 @@
 
 use hopper;
 use metric::{Encoding, Event, LogLine, Telemetry};
+use std::marker::PhantomData;
 use std::sync;
+use thread;
 use time;
 use util::Valve;
 
-mod console;
-mod firehose;
-mod null;
+pub mod console;
```

Comment:
Why mark these as pub when we're re-exporting pieces of them below?

Reply:
Uniformity.  The other modes are re-exported below and pub as well.

Reply:
I guess my question is about those, too. Originally I didn't expose these modules so we could be free to fiddle with them, choosing to export only the key types starting on line 22 / 25. 

We should choose one or the other. Either the inner modules are public or the key things are exported.

Reply:
Let's just leave them private then.  `src/source/mod.rs` leaves them private.

Reply:
So...we can't make all of these private.

The internal source currently reaches into these various modules to do its job. :(.

Reply:
Should we save making these private for a separate PR?

Reply:
Oh, I see. Hmm, maybe let's just put it back like it was and drop a ticket into the issue tracker for resolution, marked Help Wanted and Easy?

Reply:
Reverted.